packages/builder/src/lib/script-checkpoints.ts

```diff
@@ -1,105 +1,310 @@
+  }
+
+  registerASA (networkName: string, name: string, creator: string): CheckpointData {
+    registerASA(this._ensureNet(this.visibleCP, networkName), name, creator);
+    registerASA(this._ensureNet(this.strippedCP, networkName), name, creator);
+    registerASA(this._ensureNet(this.globalCP, networkName), name, creator);
+    return this;
+  }
+
+  registerASC (networkName: string, name: string, creator: string): CheckpointData {
+    registerASC(this._ensureNet(this.visibleCP, networkName), name, creator);
+    registerASC(this._ensureNet(this.strippedCP, networkName), name, creator);
+    registerASC(this._ensureNet(this.globalCP, networkName), name, creator);
+    return this;
+  }
+
+  isDefined (networkName: string, name: string): boolean {
+    const netCP = this.globalCP[networkName];
+    return netCP !== undefined &&
+      (netCP.asa[name] !== undefined || netCP.asc[name] !== undefined);
+  }
 }
 
-export function persistCheckpoint (scriptName: string, checkpoint: ScriptCheckpoint): void {
+export function persistCheckpoint (scriptName: string, checkpoint: ScriptCheckpoints): void {
   const scriptPath = toCheckpointFileName(scriptName);
   const scriptDir = path.dirname(scriptPath);
-  if (!fs.existsSync(scriptDir)) {
-    fs.mkdirSync(scriptDir, { recursive: true });
-  }
+  fs.mkdirSync(scriptDir, { recursive: true });
   fs.writeFileSync(
     scriptPath,
     YAML.stringify(checkpoint)
   );
 }
 
-export function loadCheckpoint (scriptName: string): ScriptCheckpoint {
-  const checkpointPath = toCheckpointFileName(scriptName);
-  if (!fs.existsSync(checkpointPath)) {
+export function loadCheckpointNoSuffix (checkpointPath: string): ScriptCheckpoints {
+  // Try-catch is the way:
+  // https://nodejs.org/docs/latest/api/fs.html#fs_fs_stat_path_options_callback
+  // Instead, user code should open/read/write the file directly and
+  // handle the error raised if the file is not available
+  try {
+    return YAML.parse(fs.readFileSync(checkpointPath).toString());
+  } catch (e) {
     return {};
   }
-  return YAML.parse(fs.readFileSync(checkpointPath).toString());
 }
 
+export function loadCheckpoint (scriptName: string): ScriptCheckpoints {
+  return loadCheckpointNoSuffix(toCheckpointFileName(scriptName));
+}
+
+function walk (directoryName: string): string[] {
+  var list: string[] = [];
+  fs.readdirSync(directoryName).forEach(file => {
+    var fullPath = path.join(directoryName, file);
+    const f = fs.statSync(fullPath);
+    if (f.isDirectory()) {
+      list = list.concat(walk(fullPath));
+    } else {
+      list.push(fullPath);
+    }
+  });
+  return list;
+};
+
+function findCheckpointsRecursive (): string[] {
+  const checkpointsPath = path.join(".", artifactsPath, scriptsDirectory);
+  fs.mkdirSync(checkpointsPath, { recursive: true });
+  return walk(checkpointsPath)
+    .filter(filename => filename.endsWith(checkpointFileSuffix));
+}
+
+export function loadCheckpointsRecursive (): CheckpointData {
```

Comment:
I'm not sure if `recuriseve` is a right suffix here (and in the function above).

We don't do an arbitrary recursion. We load all checkpoints. So, how about using suffix `All`?

Reply:
This loads all directories in `artifacts/scripts/` and does the recursion walk. I did it just in case.
So current title is more accurate.